### PR TITLE
fix: [FFM-3308]: Fix for add Prerequisites modal dropdown missing options

### PR DESCRIPTION
**Pull request by laura-mannus**
##### Summary:
Added an undefinied check in the logic which was causing the dropdown not appear on first click.

##### Jira Links:

](https://harness.atlassian.net/browse/FFM-3308)

##### Screenshots:

https://user-images.githubusercontent.com/103498167/168086753-b51b7a33-eace-44fb-b6a3-6bd22f67b898.mov

You can use the following comments to re-trigger PR Checks

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier - `fix prettier`

### CODE REVIEW

**Comment by laura-mannus** [comment]
retrigger titlecheck

**Comment by laura-mannus** [comment]
retrigger jest

**Review by chrisblakely01 — COMMENTED** [review]
Nice fix, might be worth adding a few unit tests in, to assert the list appears/is populated with the right data, and that the input populates with the selection on click - what ya think?

**Review by crystal-fong — APPROVED** [review]
good looking set of tests!

File: src/modules/75-cf/components/FlagActivation/__tests__/FlagPrerequisites.test.tsx [hunk 1]
```diff
@@ -55,4 +55,61 @@ describe('FlagPrerequisites', () => {
...
+    userEvent.click(screen.getByText('cf.featureFlags.newPrerequisite'))
+    await waitFor(() => expect(screen.getByTestId('flag-prerequisite-modal')).toBeInTheDocument())
+    userEvent.click(screen.getByTestId('prerequisites'))
+
+    await waitFor(() => expect(screen.getByTestId('prerequisites-form')).toBeInTheDocument())
+
+    expect(screen.getByTestId('prerequisites-variations-dropdown-0')).toBeInTheDocument()
+    userEvent.click(screen.getByTestId('prerequisites-dropdown-0'))
+    mockFeature.prerequisites?.forEach(prerequisite => {
+      expect(screen.getByText(prerequisite.feature))
```
Comment (chrisblakely01) [inline src/modules/75-cf/components/FlagActivation/__tests__/FlagPrerequisites.test.tsx:89]: should this be asserting something? e.g:

```suggestion
      expect(screen.getByText(prerequisite.feature)).toBeInTheDocument()
```

?

File: src/modules/75-cf/components/FlagActivation/__tests__/FlagPrerequisites.test.tsx [hunk 2]
```diff
@@ -55,4 +55,61 @@ describe('FlagPrerequisites', () => {
...
+      expect(screen.getByText('cf.featureFlags.newPrerequisite')).toBeInTheDocument()
+    })
+
+    userEvent.click(screen.getByText('cf.featureFlags.newPrerequisite'))
+    await waitFor(() => expect(screen.getByTestId('flag-prerequisite-modal')).toBeInTheDocument())
+    userEvent.click(screen.getByTestId('prerequisites'))
+
+    await waitFor(() => expect(screen.getByTestId('prerequisites-form')).toBeInTheDocument())
+    await waitFor(() => expect(screen.getByTestId('prerequisites-dropdown-0')).toBeInTheDocument())
+    await waitFor(() => expect(screen.getByTestId('prerequisites-variations-dropdown-0')).toBeInTheDocument())
```
Comment (chrisblakely01) [inline src/modules/75-cf/components/FlagActivation/__tests__/FlagPrerequisites.test.tsx:105]: preference thing but you can put these in a single `await waitFor()` - just as an FYI for future 👍

File: src/modules/75-cf/components/FlagActivation/FlagPrerequisites.tsx [hunk 3]
```diff
@@ -319,6 +323,7 @@ export const FlagPrerequisites: React.FC<FlagPrerequisitesProps> = props => {
                             intent="primary"
                             text={getString('cf.shared.prerequisites')}
                             icon="small-plus"
+                            data-testid="prerequisites"
```
Comment (chrisblakely01) [inline src/modules/75-cf/components/FlagActivation/FlagPrerequisites.tsx:326]: no need to change but I often like to affix what the thing is to the end of the string e.g

```suggestion
data-testid="prerequisites-button"
```

just makes it a bit easier to follow the tests at a glance - just something to consider going forward

**Review by chrisblakely01 — APPROVED** [review]
Looks good - I've one question but if non issue then happy enough 👍

File: src/modules/75-cf/components/FlagActivation/__tests__/FlagPrerequisites.test.tsx (hunk 1 above)
Comment (laura-mannus) [reply src/modules/75-cf/components/FlagActivation/__tests__/FlagPrerequisites.test.tsx:89]: Added this in :)

File: src/modules/75-cf/components/FlagActivation/__tests__/FlagPrerequisites.test.tsx (hunk 2 above)
Comment (laura-mannus) [reply src/modules/75-cf/components/FlagActivation/__tests__/FlagPrerequisites.test.tsx:105]: Understandable, I have replaced :)

File: src/modules/75-cf/components/FlagActivation/FlagPrerequisites.tsx (hunk 3 above)
Comment (laura-mannus) [reply src/modules/75-cf/components/FlagActivation/FlagPrerequisites.tsx:326]: Totally agree have changed, thank you

File: src/modules/75-cf/components/FlagActivation/__tests__/FlagPrerequisites.test.tsx [hunk 4]
```diff
@@ -55,4 +55,63 @@ describe('FlagPrerequisites', () => {
...
+    renderComponent()
+    await waitFor(() => {
+      expect(screen.getByText('cf.featureFlags.newPrerequisite')).toBeInTheDocument()
+    })
+
+    userEvent.click(screen.getByText('cf.featureFlags.newPrerequisite'))
+
+    await waitFor(() => expect(screen.getByTestId('flag-prerequisite-modal')).toBeInTheDocument())
+
+    userEvent.click(screen.getByTestId('prerequisites-button'))
```
Comment (knagurski) [inline src/modules/75-cf/components/FlagActivation/__tests__/FlagPrerequisites.test.tsx:69]: Try to use `getByRole` instead, so we keep our tests based on what the user can see and interact with.

File: src/modules/75-cf/components/FlagActivation/__tests__/FlagPrerequisites.test.tsx [hunk 5]
```diff
@@ -55,4 +55,63 @@ describe('FlagPrerequisites', () => {
 
     expect(screen.getByTestId('flag-prerequisite-modal')).toMatchSnapshot()
   })
+
+  test('form should render when adding prerequisite', async () => {
+    renderComponent()
+    await waitFor(() => {
+      expect(screen.getByText('cf.featureFlags.newPrerequisite')).toBeInTheDocument()
+    })
```
Comment (knagurski) [inline src/modules/75-cf/components/FlagActivation/__tests__/FlagPrerequisites.test.tsx:63]: Does this need a `waitFor`? Typically you only need these after some kind of interaction, not the initial render.

File: src/modules/75-cf/components/FlagActivation/__tests__/FlagPrerequisites.test.tsx [hunk 6]
```diff
@@ -55,4 +55,63 @@ describe('FlagPrerequisites', () => {
...
+
+    userEvent.click(screen.getByText('cf.featureFlags.newPrerequisite'))
```
Comment (knagurski) [inline src/modules/75-cf/components/FlagActivation/__tests__/FlagPrerequisites.test.tsx:65]: Is this a button or a link? If so, try `getByRole` instead.

File: src/modules/75-cf/components/FlagActivation/__tests__/FlagPrerequisites.test.tsx [hunk 7]
```diff
@@ -55,4 +55,63 @@ describe('FlagPrerequisites', () => {
...
+    await waitFor(() => expect(screen.getByTestId('prerequisites-form')).toBeInTheDocument())
+
+    expect(screen.getByTestId('prerequisites-variations-dropdown-0')).toBeInTheDocument()
+    userEvent.click(screen.getByTestId('prerequisites-dropdown-0'))
+    mockFeature.prerequisites?.forEach(prerequisite => {
+      expect(screen.getByText(prerequisite.feature)).toBeInTheDocument()
+    })
+  })
+
+  test('its should render variation when feature flag is selected', async () => {
```
Comment (knagurski) [inline src/modules/75-cf/components/FlagActivation/__tests__/FlagPrerequisites.test.tsx:93]:
```suggestion
  test('it should render variation when feature flag is selected', async () => {
```

File: src/modules/75-cf/components/FlagActivation/__tests__/FlagPrerequisites.test.tsx [hunk 8]
```diff
@@ -55,4 +55,63 @@ describe('FlagPrerequisites', () => {
...
+
+    await waitFor(() => {
+      expect(screen.getByTestId('prerequisites-form')).toBeInTheDocument()
+      expect(screen.getByTestId('prerequisites-dropdown-0')).toBeInTheDocument()
+      expect(screen.getByTestId('prerequisites-variations-dropdown-0')).toBeInTheDocument()
+    })
+
+    mockFeature.prerequisites?.forEach(prerequisite => {
+      userEvent.click(screen.getByTestId('prerequisites-dropdown-0'))
+      userEvent.dblClick(screen.getByText(prerequisite.feature))
```
Comment (knagurski) [inline src/modules/75-cf/components/FlagActivation/__tests__/FlagPrerequisites.test.tsx:111]: Why double click?

**Review by knagurski — APPROVED** [review]
Nice simple fix, but some questions/suggestions around the tests.

**Pull request merged**